bot/exts/evergreen/connect_four.py

```diff
@@ -0,0 +1,440 @@
+        else:
+            self.message = await self.channel.send(content='Loading ....')
+            for emoji in self.unicode_numbers:
+                await self.message.add_reaction(emoji)
+            await self.message.add_reaction(CROSS_EMOJI)
+            await self.message.edit(content=None, embed=embed)
+
+    async def game_over(self, winner: discord.user, loser: discord.user) -> None:
+        """Removes games from list of current games and announces to public chat."""
+        await self.channel.send(f"Game Over! {winner.mention} won against {loser.mention}")
+        await self.print_grid()
+
+    async def start_game(self) -> None:
+        """Begins the game."""
+        self.player_active, self.player_inactive = self.player1, self.player2
+
+        while True:
+            await self.print_grid()
+
+            if isinstance(self.player_active, AI):
+                coords = self.player_active.play()
+                if not coords:
+                    await self.channel.send("Game Over! It's A Draw :tada:")
+                    await self.print_grid()
+            else:
+                coords = await self.player_turn()
+
+            if not coords:
+                return
+
+            if self.check_win(coords, 1 if self.player_active == self.player1 else 2):
+                await self.game_over(
+                    self.bot.user if isinstance(self.player_active, AI) else {self.player_active},
+                    self.bot.user if isinstance(self.player_inactive, AI) else {self.player_inactive},
+                )
+                return
+
+            self.player_active, self.player_inactive = self.player_inactive, self.player_active
+
+    def predicate(self, reaction: discord.Reaction, user: discord.Member) -> bool:
+        """The predicate to check for the player's reaction."""
+        return (
+            reaction.message.id == self.message.id
+            and user.id == self.player_active.id
+            and str(reaction.emoji) in (*self.unicode_numbers, CROSS_EMOJI)
+        )
+
+    async def player_turn(self) -> Coordinate:
+        """Initiate the player's turn."""
+        message = await self.channel.send(
+            f"{self.player_active.mention}, it's your turn! React with the column you want to place your token in."
+        )
+        player_num = 1 if self.player_active == self.player1 else 2
+        while True:
+            try:
+                reaction, user = await self.bot.wait_for("reaction_add", check=self.predicate, timeout=30.0)
+            except asyncio.TimeoutError:
+                await self.channel.send(f"{self.player_active.mention}, you took too long. Game over!")
+                return
+            else:
+                if str(reaction.emoji) == CROSS_EMOJI:
+                    await message.delete()
+                    await self.channel.send(f"{self.player_active.user} surrendered. Game over!")
+                    return
+
+                await message.delete()
```

Comment:
This can be palced before the if statement above, so you don't need to repeat it in two spots.